### ISSUE TITLE
hadolint suggests combining RUN instructions with heredoc

### ISSUE THREAD

**Issue by rittneje** [hadolint/hadolint#805]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior

Hadolint should not suggest combining `RUN` instructions when one of them involves a heredoc, as there does not seem to be any way to actually combine these instructions properly.

### Actual behavior

It spits out a violation: "Multiple consecutive `RUN` instructions. Consider consolidation."

### Steps to reproduce the behavior

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

```bash
Haskell Dockerfile Linter v2.9.3-0-g346e419-dirty
```

**Dockerfile** (if relevant)

```Dockerfile
# syntax=docker.io/docker/dockerfile:labs

FROM ubuntu:focal
RUN cat <<EOF
Here is a heredoc.
EOF

RUN echo 'there'
```

**Additional environment details (OS, stack version, etc.)**

**Comment by rittneje** [comment]
After lots of experimentation, I found this can be resolved with a nested heredoc. But this is not at all obvious. DL3059 should definitely output something more useful.

```Dockerfile
FROM ubuntu:focal
RUN bash <<EOF
cat <<END
Here is a heredoc.
END
echo 'there'
EOF
```
